codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -231,9 +255,22 @@ private int connectionWindowSize() {
     }
 
     /**
+     * Determine if the underlying channel is writable.
+     */
+    private boolean isWritable() {
+        return ctx != null && ctx.channel().isWritable();
+    }
+
+    /**
      * Writes as many pending bytes as possible, according to stream priority.
+     * <p>
+     * This method will only take action if {@link #isWritable()} is {@code true}.
      */
     private void writePendingBytes() throws Http2Exception {
+        if (!isWritable()) {
```

Comment:
Yes. I think the `writabilityChanged()` method provides what your looking for?

https://github.com/netty/netty/pull/3864/files#diff-d8ca391b616f0c1db6e8f345ca3284feR230